### PR TITLE
corp: add duration support

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @qiuyesuifeng @coocood  PTAL

### CODE REVIEW

File: src/util/codec/mysql/duration.rs [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+fn tm_to_secs(t: Tm) -> u64 {
+    t.tm_hour as u64 * SECS_PER_HOUR + t.tm_min as u64 * SECS_PER_MINUTE + t.tm_sec as u64
+}
+
+/// `Duration` is the type for `MySQL` time type.
+#[derive(Debug, Clone)]
+pub struct Duration {
+    pub dur: StdDuration,
+    past: bool,
```
Comment (coocood) [inline src/util/codec/mysql/duration.rs]: `negative` should be better.
`past` doesn't necessarily mean the duration is negative, for example, a duration from yesterday to today is a past duration, but it is still positive.

File: src/util/codec/mysql/duration.rs [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+        let res = self.dur.as_secs() as f64 + self.dur.subsec_nanos() as f64 * 10e-9;
+        if self.past {
+            -res
+        } else {
+            res
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.dur.as_secs() == 0 && self.dur.subsec_nanos() == 0
```
Comment (coocood) [inline src/util/codec/mysql/duration.rs]: Is there a total nano seconds function? so you just need to compare once.

**Comment by coocood** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**